.teamcity/settings.kts

```diff
@@ -1322,24 +1330,29 @@ object WPComPlugins_EditorToolKit : BuildType({
 				export NODE_ENV="production"
 
 				cd apps/editing-toolkit
+				yarn build
 
-				# Update plugin version in the plugin file.
-				sed -i -e "/^\s\* Version:/c\ * Version: %build.number%" -e "/^define( 'A8C_ETK_PLUGIN_VERSION'/c\define( 'A8C_ETK_PLUGIN_VERSION', '%build.number%' );" ./editing-toolkit-plugin/full-site-editing-plugin.php
+				# Update plugin version in the plugin file and readme.txt.
+				# Note: we also update the latest build from trunk to this version to restore idempotence
+				sed -i -e "/^\s\* Version:/c\ * Version: %build.number%" -e "/^define( 'A8C_ETK_PLUGIN_VERSION'/c\define( 'A8C_ETK_PLUGIN_VERSION', '%build.number%' );" ./editing-toolkit-plugin/full-site-editing-plugin.php ../../editing-toolkit-last-build/full-site-editing-plugin.php
+				sed -i -e "/^Stable tag:\s/c\Stable tag: %build.number%" ./editing-toolkit-plugin/readme.txt ../../editing-toolkit-last-build/readme.txt
 
-				# Update plugin stable tag in readme.txt.
-				sed -i -e "/^Stable tag:\s/c\Stable tag: %build.number%" ./editing-toolkit-plugin/readme.txt
+				if diff -rq ./editing-toolkit-plugin/ ../../editing-toolkit-last-build/ ; then
+					echo "The build matches trunk. Therefore, this build has no effect."
+				else
+					echo "The build is different from trunk."
```

Comment:
We basically only publish an artifact when something has changed. For example, in this build, the output was the same as the existing trunk build, so it did not publish an artifact: https://teamcity.a8c.com/buildConfiguration/calypso_WPComPlugins_EditorToolKit/5922379?buildTab=artifacts